### PR TITLE
Rahul/ifl 2271 add signing commitment

### PR DESCRIPTION
**Pull request by patnir**
Continues the path of moving structs from ironfish_utils

### CODE REVIEW

**Review by andiflabs — APPROVED** [review]

File: src/signing_commitment.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+use crate::participant::{Identity, IDENTITY_LEN};
+
+const SIGNING_COMMITMENT_LENGTH: usize = IDENTITY_LEN + 96;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct SigningCommitment {
+    pub identity: Identity,
+    pub hiding: NonceCommitment,
+    pub binding: NonceCommitment,
```
Comment (hughy) [inline src/signing_commitment.rs]: should these be public?

File: src/signing_commitment.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::io;
+
+use reddsa::frost::redjubjub::round1::NonceCommitment;
+
+use crate::participant::{Identity, IDENTITY_LEN};
+
+const SIGNING_COMMITMENT_LENGTH: usize = IDENTITY_LEN + 96;
```
Comment (hughy) [inline src/signing_commitment.rs]: why + 96? from `deserialize_from` it looks like the hiding and binding commitments are each 32 bytes

File: src/signing_commitment.rs (hunk 1 above)
Comment (patnir) [reply src/signing_commitment.rs]: good catch, removed the pub

File: src/signing_commitment.rs (hunk 2 above)
Comment (patnir) [reply src/signing_commitment.rs]: changed this to reflect the correct number. I believe it is an error introduced here https://github.com/iron-fish/ironfish/pull/4744/files

**Pull request merged**